pkg/controller/storagecluster/storagecluster.go

```diff
@@ -447,8 +448,10 @@ func (c *Controller) runPreflightCheck(cluster *corev1.StorageCluster) error {
 		err = c.client.List(context.TODO(), k8sNodeList)
 		if err == nil {
 			for _, node := range k8sNodeList.Items {
-				logrus.Infof("Create pre-flight storage node entry for node: %s", node.Name)
-				c.createStorageNode(cluster, node.Name)
+				if !coreops.Instance().IsNodeMaster(node) {
```

Comment:
@jrivera-px what's still not clear to me is - How is it that the list contains the master node sometimes and not always?

Reply:
The list always contained the master node.   In the past the checks we made did not consider the it a failed validation if the master node checks was nil.